### ISSUE TITLE
DetectnNet part missing

### ISSUE THREAD

**Issue by princefr** [mks0601/3DMPPE_POSENET_RELEASE#5]
Hi thank you for your work guys, do you gonna open source the DetectNet part ?
Regards

**Comment by mks0601** [comment]
Our detectnet is based one mask rcnn repo (https://github.com/facebookresearch/maskrcnn-benchmark). Bounding box prediction results are provided in README. Do you want source code of DetectNet or want bounding box prediction?

**Comment by princefr** [comment]
Hi thank you for taking time to answer i appreciate it, i'm trying to rewrite your paper with all the parts, detectnet, rootnet, and posenet all together i was wondering if the detectnet part was just a normal maskrcnn pipeline or an in house one made by you guys. if its the case (an in house one ) it would be great to open source it if its just the normal one don't loose your time posting it. 

Regards

**Comment by mks0601** [comment]
We didn't change network architecture from the Mask R-CNN. I guess you can just use that repo for your purpose.

**Comment by wangzheallen** [comment]
Hi @mks0601 , could you please specify which model file you use for testing in https://github.com/facebookresearch/maskrcnn-benchmark? 

Thanks!

**Comment by mks0601** [comment]
I used resnet-50,101,X-101 mask r-cnn as described in my paper (https://arxiv.org/abs/1907.11346). Also, as you can see in my paper, human detection performance has marginal effect on final 3D human pose estimation accuracy.

**Comment by wangzheallen** [comment]
> I used resnet-50,101,X-101 mask r-cnn as described in my paper (https://arxiv.org/abs/1907.11346). Also, as you can see in my paper, human detection performance has marginal effect on final 3D human pose estimation accuracy.

Thanks for the reply, so you do not use 'FPN'? and it seems the RootNet is influenced a lot.

**Comment by mks0601** [comment]
I used FPN. RootNet is different module

**Comment by wangzheallen** [comment]
what is the difference in terms of usage between image bbox projected by 2m*2m*2m and bbox sanitized by 2d joint location?

**Comment by mks0601** [comment]
Could you clarify your question? what do you want to get?

**Comment by mks0601** [comment]
Obviously, bbox projected by 2m 2m 2m can be different from extended bounding box from 2d joint locations

**Comment by wangzheallen** [comment]
> Obviously, bbox projected by 2m 2m 2m can be different from extended bounding box from 2d joint locations

in https://github.com/mks0601/3DMPPE_POSENET_RELEASE/blob/master/data/Human36M/Human36M.py#L117
you sanitize the bbox, [bbox1]
in https://github.com/mks0601/3DMPPE_POSENET_RELEASE/blob/master/data/dataset.py#L58
you uses the bbox to crop the image, [bbox2]
how to get bbox1 and bbox2, and what are they used for?

Thanks!

**Comment by mks0601** [comment]
bbox2 is the same with bbox1

**Comment by wangzheallen** [comment]
> bbox2 is the same with bbox1

Thanks for the reply. May I ask whether the bbox
https://github.com/mks0601/3DMPPE_POSENET_RELEASE/blob/master/data/Human36M/Human36M.py#L121
comes from projection of (2m*2m*2m)3d bbox?

**Comment by mks0601** [comment]
No, it comes from 2d joint locations. Can I ask why do you think the bbox1 is from projection?

**Comment by wangzheallen** [comment]
> No, it comes from 2d joint locations. Can I ask why do you think the bbox1 is from projection?

I was thinking when training the cropped image [bbox3] from projected 3d bbox, but testing [bbox1] it can be from 2d joint location and sanitized bbox. 

As the paper mentioned, the bbox is from Mask-RCNN and when you use the pinhole camera model,  you use the area(bbox), so I assume during test time it from Mask-RCNN and during training time, you use the 2m*2m*2m 3d bbox and projected to 2d to get the bbox.

It turns out for the detectNet you use ground truth 2d joint (training time) or Mask-RCNN detected 2d joint (test time), and use the 2d joint to get initial bbox, and then sanitize them. And use the sanitized bbox to crop image for RootNet or PoseNet, am I correct?

**Comment by mks0601** [comment]
Let me clarify which bounding boxes are used for three modules (i.e., DetectNet, PoseNet, and RootNet).

1. DetectNet
Training: pre-trained on MS COCO dataset. No fine-tuning on 3D human pose estimation dataset.
Test: Bounding box prediction result on 3D human pose estimation dataset from pre-trained Mask R-CNN

2. PoseNet (When cropping each human in the input image)
Training: Bounding box generated and extended from 2D joint locations
Test: Bounding box from the DetectNet

3. RootNet (When cropping each human in the input image, when calculating A_{img} for k value)
Training: Bounding box generated and extended from 2D joint locations
Test: Bounding box from the DetectNet

**Comment by wangzheallen** [comment]
> Let me clarify which bounding boxes are used for three modules (i.e., DetectNet, PoseNet, and RootNet).
> 
>     1. DetectNet
>        Training: Pre-trained on MS COCO dataset. No fine-tuning on 3D human pose estimation dataset.
>        Test: Bounding box prediction result on 3D human pose estimation dataset from pre-trained Mask R-CNN
> 
>     2. PoseNet (When cropping each human in the input image)
>        Training: Bounding box generated and extended from groundtruth 2D joint locations
>        Test: Bounding box from the DetectNet
> 
>     3. RootNet (When cropping each human in the input image, when calculating A_{img} for k value)
>        Training: Bounding box generated and extended from groundtruth 2D joint locations
>        Test: Bounding box from the DetectNet

Thanks! everything clear now!
